addons/project/models/project.py

```diff
@@ -1006,6 +1006,7 @@ def message_subscribe(self, partner_ids=None, channel_ids=None, subtype_ids=None
         if partner_ids:
             new_allowed_users = self.env['res.partner'].browse(partner_ids).user_ids.filtered('share')
             tasks = self.filtered(lambda task: task.project_id.privacy_visibility == 'portal')
+            tasks.allowed_user_ids  # ensure field is computed before modifying it
             tasks.sudo().write({'allowed_user_ids': [(4, user.id) for user in new_allowed_users]})
```

Comment:
What about this?
I find this more elegant than calling the field value before the write.
I might be wrong but I think it will do the exact same thing. What do you think @nle-odoo ?
Thanks for your help guys 🙂 
```suggestion
            tasks.sudo().allowed_user_ids |= user
```

Reply:
```suggestion
            tasks.allowed_user_ids  # ensure field is computed before modifying it
            tasks.sudo().write({'allowed_user_ids': [(4, user.id) for user in new_allowed_users]})
            tasks.sudo().allowed_user_ids |= new_allowed_users
```

Reply:
It passes the test

Reply:
> So it would be?:

Indeed, sorry!

Reply:
looks good